ONSdigital/dp-renderer
# Feature/date validation error enhancements

### What

- Added `bool` for day/month/year to display css class `ons-input--error` when field has validation error 
- Created date fieldset that encapsulates field validation and associated model
- Updated README with new fieldset

### How to review

- Sense check
- Tests pass
- Pull this branch and following these instructions:-
  - Update `go.mod` of your favourite frontend app (I used `dp-frontend-release-calendar`) to use local `dp-renderer` instance i.e. `replace "github.com/ONSdigital/dp-renderer/v2" => "/src/github.com/ONSdigital/dp-renderer"`
  -  Add `NewDate             coreModel.DateFieldset` to the model
  - Add template reference `{{ template "partials/fields/fieldset-date" .NewDate }}`
  - Add this snippet in `mapper.go`
```go
calendar.NewDate = coreModel.DateFieldset{
		ErrorID: "new-date-error",
		Input: coreModel.InputDate{
			Language:            lang,
			Id:                  "new-date",
			InputNameDay:        "new-day",
			InputNameMonth:      "new-month",
			InputNameYear:       "new-year",
			InputValueDay:       params.AfterDate.DayString(),
			InputValueMonth:     params.AfterDate.MonthString(),
			InputValueYear:      params.AfterDate.YearString(),
			HasDayValidationErr: true,
			Title: coreModel.Localisation{
				Text: "New date title",
			}, Description: coreModel.Localisation{
				Text: "Date description",
			}},
		Language: lang,
		ValidationErrDescription: []coreModel.Localisation{
			{
				Text: "Oops",
			},
			{
				Text: "Whoops",
			},
		},
	}
```

<img width="392" alt="new date fieldset with validation error" src="https://github.com/ONSdigital/dp-renderer/assets/19624419/0686c3c5-4d69-451d-a704-7ac977c6919f">


### Who can review

Frontend go dev
